src/routes.ts

```diff
@@ -2,9 +2,17 @@ import Router from "@koa/router";
 import Koa from "koa";
 import { ApiGatewayContext } from "./middleware";
 import createError from "http-errors";
-import getCharacterMetadata from "./api/getCharacterMetadata";
-import { getCharacterSchema } from "./api/validators";
+import { getCharacterSchema, getMatchesSchema } from "./api/validators";
 import { requestCrawl } from "./lib/sqs/sqs_producer";
+import {
+  checkCharacterExists,
+  getCharacterProfile,
+} from "./lib/warmane_client/client";
+import {
+  characterMetaStore,
+  crawlerStateStore,
+  matchDetailsStore,
+} from "./db/documentStoreV2";
 
```

Comment:
As far as I can tell, if the SQS crawl request fails for any reason, crawler state would remain as `pending` even though the request wasn't successful. Aside from misleading users, this could be a cascading issue if we were to tie any services to this state. I actually have a possible solution for this one:
  - Wrap `requestCrawl` in a try/catch block.
  - If an error is caught, reset the crawler state.

Reply:
I think it is in a try/catch block at the moment and it writes an errored state to Dynamo in the case that it ever fails.

You're right though in that getting stuck in a particular state is a very real possibility and could be problematic.

Here is another potential issue that I thought of. Consider the fact that Dynamo is inherently _eventually_ consistent. This means that immediate reads after writes aren't guaranteed to return what you just wrote to the database.

Suppose I write "pending" to Dynamo, and submit a request for the crawler to start. If Dynamo hasn't fully written "pending" when the crawler reads the current state, it might get back "idle" or `undefined`. I would conjecture that at our small scale we will never observe this, it is still possible.